### PR TITLE
chore: create templated getSpecificHandler - no need to use repetitiv…

### PR DESCRIPTION
**Pull request by JakubFornadel**
## Purpose

Note: just a quicker way how to explain this: https://github.com/Taraxa-project/taraxa-node/pull/1716#discussion_r872120851

### CODE REVIEW

File: libraries/core_libs/network/include/network/tarcap/packets_handler.hpp [hunk 1]
```diff
@@ -35,4 +43,9 @@ class PacketsHandler {
   std::unordered_map<SubprotocolPacketType, std::shared_ptr<PacketHandler>> packets_handlers_;
 };
 
+template <typename PacketHandlerType>
+std::shared_ptr<PacketHandlerType> PacketsHandler::getSpecificHandler(SubprotocolPacketType packet_type) {
+  return std::static_pointer_cast<PacketHandlerType>(getSpecificHandler(packet_type));
+}
```
Comment (MatusKysel) [inline libraries/core_libs/network/include/network/tarcap/packets_handler.hpp:49]: would it be possible to have some kind of table so we do not need to specify both? packet type and packet type handler? or maybe `PacketHandlerType` could have some kind of static variable or not sure how it is handled and based on that we would get the type? `PacketHandlerType::packet_type` ???

**Review by MatusKysel — APPROVED** [review]

**Review by kstdl — APPROVED** [review]

File: libraries/core_libs/network/include/network/tarcap/packets_handler.hpp (hunk 1 above)
Comment (JakubFornadel) [reply libraries/core_libs/network/include/network/tarcap/packets_handler.hpp:49]: yes I was thinking about it too, there could be a big switch and thats about it - not sure what is better solution, have this big switch or just specify the type ?

**Review by mfrankovi — APPROVED** [review]

File: libraries/core_libs/network/include/network/tarcap/packets_handler.hpp (hunk 1 above)
Comment (JakubFornadel) [reply libraries/core_libs/network/include/network/tarcap/packets_handler.hpp]: @MatusKysel @kstdl fixed, please re - review :)

**Review by kstdl — APPROVED** [review]

File: libraries/core_libs/network/include/network/tarcap/packets_handler.hpp [hunk 2]
```diff
@@ -20,19 +20,43 @@ class PacketsHandler {
...
+  void registerHandler(Args&&... args);
 
  private:
   // Map of all packets handlers, factory method selects specific packet handler for processing based on packet type
   std::unordered_map<SubprotocolPacketType, std::shared_ptr<PacketHandler>> packets_handlers_;
 };
 
+template <typename PacketHandlerType>
+std::shared_ptr<PacketHandlerType> PacketsHandler::getSpecificHandler() {
+  return std::static_pointer_cast<PacketHandlerType>(getSpecificHandler(PacketHandlerType::k_packet_type));
```
Comment (MatusKysel) [inline libraries/core_libs/network/include/network/tarcap/packets_handler.hpp:52]: hmm so are we agreed on k_something_something ?

Comment (JakubFornadel) [reply libraries/core_libs/network/include/network/tarcap/packets_handler.hpp:52]: Aah I forgot, (yeah I still think its bullshit to have different styles for variables :D) but you guys decided on slack poll so democracy it is - will refactor style...

**Review by gaoqi8649 — APPROVED** [review]

**Pull request merged**